homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,542 @@
+                                                       mime_type,
+                                                       upnp_class)
+            await self._device.async_wait_for_can_play()
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Error during SetAVTransportURI or wait call")
+            return
+
+        # if already playing, no need to call Play
+        from async_upnp_client import dlna
+        if self._device.state == dlna.STATE_PLAYING:
+            return
+
+        # play it
+        await self.async_media_play()
+
+    async def async_media_previous_track(self):
+        """Send previous track command."""
+        if not self._device.can_previous:
+            _LOGGER.debug('Cannot do Previous')
+            return
+
+        try:
+            await self._device.async_previous()
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Error during Previous call")
+
+    async def async_media_next_track(self):
+        """Send next track command."""
+        if not self._device.can_next:
+            _LOGGER.debug('Cannot do Next')
+            return
+
+        try:
+            await self._device.async_next()
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Error during Next call")
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._device.media_title
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._device.media_image_url
+
+    @property
+    def state(self):
+        """State of the player."""
+        if not self._available:
+            return STATE_OFF
+
+        from async_upnp_client import dlna
+        if self._device.state is None:
+            return STATE_ON
+        elif self._device.state == dlna.STATE_PLAYING:
+            return STATE_PLAYING
+        elif self._device.state == dlna.STATE_PAUSED:
+            return STATE_PAUSED
+
+        return STATE_IDLE
+
+    @property
+    def media_duration(self):
+        """Duration of current playing media in seconds."""
+        return self._device.media_duration
+
+    @property
+    def media_position(self):
+        """Position of current playing media in seconds."""
+        return self._device.media_position
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        return self._device.media_position_updated_at
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        if self._name:
+            return self._name
+        return self._device.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return an unique ID."""
+        return "{}.{}".format(__name__, self.udn)
```

Comment:
Remove `__name__` from the unique id. The core will keep track of platform and domain together with the unique id.